moveit_core/robot_model/include/moveit/robot_model/link_model.h

```diff
@@ -41,13 +41,16 @@
 #include <vector>
 #include <utility>
 #include <map>
+#include <boost/shared_ptr.hpp>
 #include <Eigen/Geometry>
 #include <eigen_stl_containers/eigen_stl_vector_container.h>
 #include <moveit/macros/class_forward.h>
 
 namespace shapes
 {
-MOVEIT_CLASS_FORWARD(Shape);
+class Shape;
+typedef boost::shared_ptr<Shape> ShapePtr;
+typedef boost::shared_ptr<const Shape> ShapeConstPtr;
```

Comment:
I didn't seem your last comment when replying. But fair enough, it does make sense to switch `geometric_shapes` and `moveit` at the same time. It could get very confusing otherwise.

And in that light it may be acceptable to use `MOVEIT_CLASS_FORWARD` for `geometric_shapes` types. It feels a bit weird to define typedefs using a moveit convention in the namespace of another library, but I suppose `geometric_shapes` and moveit are related enough.

Shall I just drop the relevant commit then, and revert to using `MOVEIT_CLASS_FORWARD`?